packages/sources/intrinio/src/util.ts

```diff
@@ -0,0 +1,296 @@
+    }
+  }
+
+  _throw(e: any) {
+    let handled = false
+    if (typeof e === 'string') {
+      e = 'IntrinioRealtime | ' + e
+    }
+    if (this.listenerCount('error') > 0) {
+      Logger.error(e)
+      handled = true
+    }
+    if (!handled) {
+      throw e
+    }
+  }
+
+  _makeAuthUrl(): { host: string; path: string } {
+    let auth_url = {
+      host: '',
+      path: '',
+    }
+
+    if (this.options.provider == 'iex') {
+      auth_url = {
+        host: 'realtime.intrinio.com',
+        path: '/auth',
+      }
+    } else if (this.options.provider == 'quodd') {
+      auth_url = {
+        host: 'api.intrinio.com',
+        path: '/token?type=QUODD',
+      }
+    } else if (this.options.provider == 'cryptoquote') {
+      auth_url = {
+        host: 'crypto.intrinio.com',
+        path: '/auth',
+      }
+    } else if (this.options.provider == 'fxcm') {
+      auth_url = {
+        host: 'fxcm.intrinio.com',
+        path: '/auth',
+      }
+    }
+
+    if (this.options.api_key) {
+      auth_url = this._makeAPIAuthUrl(auth_url)
+    }
+
+    return auth_url
+  }
+
+  _makeAPIAuthUrl(auth_url: any): any {
+    let path = auth_url.path
+
+    if (path.includes('?')) {
+      path = path + '&'
+    } else {
+      path = path + '?'
+    }
+
+    auth_url.path = path + 'api_key=' + this.options.api_key
+    return auth_url
+  }
+
+  _makeHeaders() {
+    if (this.options.api_key) {
+      return {
+        'Content-Type': 'application/json',
+      }
+    } else {
+      const { username, password } = this.options
+
+      return {
+        'Content-Type': 'application/json',
+        Authorization: 'Basic ' + Buffer.from(username + ':' + password).toString('base64'),
+      }
+    }
+  }
+
+  _refreshToken() {
+    Logger.debug('Requesting auth token...')
+
+    return new Promise<void>((fulfill, reject) => {
+      const agent = this.options.agent || false
+      const { host, path } = this._makeAuthUrl()
+      const headers = this._makeHeaders()
+
+      // Get token
+      const options = {
+        host: host,
+        path: path,
+        agent: agent,
+        headers: headers,
+      }
+
+      const req = https.get(options, (res: any) => {
+        if (res.statusCode == 401) {
+          this._throw('Unable to authorize')
+          reject()
+        } else if (res.statusCode != 200) {
+          console.error(
+            'IntrinioRealtime | Could not get auth token: Status code ' + res.statusCode,
+          )
+          reject()
+        } else {
+          res.on('data', (data: any) => {
+            this.token = Buffer.from(data, 'base64').toString()
+            Logger.debug('Received auth token!')
+            fulfill()
+          })
+        }
+      })
+
+      req.on('error', (e) => {
+        console.error('IntrinioRealtime | Could not get auth token: ' + e)
+        reject(e)
+      })
+
+      req.end()
+    })
+  }
+
+  async _makeSocketUrl(): Promise<string> {
```

Comment:
Instead of using the `_` unused syntax, we should make use of typescripts `private`, `protected`, and `public` access modifiers. If you really want it to be private, you can use the javascript private fields access modifier `#` to make it "hard" private, where its not accessible from even runtime code.

It looks like we don't need to worry about hard privacy though, but ill let you make the final judgement call on this since you have more context.

See https://www.typescriptlang.org/docs/handbook/2/classes.html#private for more details